force-app/main/default/classes/SOQL.cls

```diff
@@ -271,11 +315,12 @@ public with sharing class SOQL {
      */
     public static List<Contact> getAccountsWithContactsByState(Id accountId) {
         // Query the Account
-
+        Account acc = [SELECT Id, BillingState FROM Account WHERE Id = :accountId LIMIT 1];
         // Store the Account's BillingState
-
+        String state = acc.BillingState;
         // Query the Contacts
-        return null; // Replace null with the variable you used to store the result
+        List<Contact> cons = [SELECT Id, Name, MailingState FROM Contact WHERE MailingState = :state];
+        return cons; // Replace null with the variable you used to store the result
```

Comment:
We did not go over this, but can you find a way to do this with one query? HINT: There will be a query in your criteria